src/ray/raylet/node_manager.cc

```diff
@@ -2763,33 +2909,44 @@ void NodeManager::HandleObjectLocal(const ObjectID &object_id) {
                  << " on " << self_node_id_ << ", " << ready_task_ids.size()
                  << " tasks ready";
   // Transition the tasks whose dependencies are now fulfilled to the ready state.
-  if (ready_task_ids.size() > 0) {
-    std::unordered_set<TaskID> ready_task_id_set(ready_task_ids.begin(),
-                                                 ready_task_ids.end());
-
-    // First filter out the tasks that should not be moved to READY.
-    local_queues_.FilterState(ready_task_id_set, TaskState::BLOCKED);
-    local_queues_.FilterState(ready_task_id_set, TaskState::RUNNING);
-    local_queues_.FilterState(ready_task_id_set, TaskState::DRIVER);
-    local_queues_.FilterState(ready_task_id_set, TaskState::WAITING_FOR_ACTOR_CREATION);
-
-    // Make sure that the remaining tasks are all WAITING or direct call
-    // actors.
-    auto ready_task_id_set_copy = ready_task_id_set;
-    local_queues_.FilterState(ready_task_id_set_copy, TaskState::WAITING);
-    // Filter out direct call actors. These are not tracked by the raylet and
-    // their assigned task ID is the actor ID.
-    for (const auto &id : ready_task_id_set_copy) {
-      RAY_CHECK(actor_registry_.count(id.ActorId()) > 0);
-      ready_task_id_set.erase(id);
+  if (new_scheduler_enabled_) {
+    for (auto task_id: ready_task_ids) {
+      auto it = waiting_tasks_.find(task_id);
+      if (it != waiting_tasks_.end()) {
+        task_dependency_manager_.UnsubscribeGetDependencies(task_id);
+        tasks_to_dispatch_.push_back(it->second);
```

Comment:
Don't you need to dispatch these in case tasks got added to `tasks_to_dispatch_`?

Reply:
Good catch. Thanks